gnc/sub8_trajectory_generator/test/tgen_manager_test.cc

```diff
@@ -0,0 +1,119 @@
+/**
+ * Author: Patrick Emami
+ * Date: 9/30/15
+ */
+#include "tgen_manager.h"
+#include "tgen_common.h"
+#include "sub8_msgs/Path.h"
+#include "sub8_state_space.h"
+#include "ompl/base/StateSampler.h"
+#include <sub8_alarm/alarm_helpers.h>
+#include <sub8_msgs/PathPoint.h>
+#include <Eigen/Geometry>
+#include <Eigen/Dense>
+#include <gtest/gtest.h>
+#include <boost/math/constants/constants.hpp>
+
+using sub8::trajectory_generator::TGenManager;
+using sub8::trajectory_generator::TGenManagerPtr;
+using sub8::trajectory_generator::Sub8StateSpace;
+using sub8::AlarmBroadcasterPtr;
+using sub8::AlarmBroadcaster;
+using ompl::base::StateSamplerPtr;
+
+class TGenManagerTest : public ::testing::Test {
+ public:
+  TGenManagerTest()
+      : _node_handle(new ros::NodeHandle()),
+        _test_space(new Sub8StateSpace()){};
+
+  boost::shared_ptr<ros::NodeHandle> getNodeHandle() { return _node_handle; }
+
+  TGenManagerPtr tgenManagerFactory() {
+    AlarmBroadcasterPtr ab(new AlarmBroadcaster(getNodeHandle()));
+    return TGenManagerPtr(new TGenManager(std::move(ab)));
+  }
+
+  const ompl::base::StateSpacePtr& ss() const { return _test_space; }
+
+ private:
+  boost::shared_ptr<ros::NodeHandle> _node_handle;
+  ompl::base::StateSpacePtr _test_space;
+};
+
+TEST_F(TGenManagerTest, testSetProblemDefinition) {
```

Comment:
This is just a test fixture for creating an instance of the TGenManager class for each test